src/main/java/mate/academy/intro/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+package mate.academy.intro.service.impl;
+
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.intro.dto.CreateOrderRequestDto;
+import mate.academy.intro.dto.OrderItemResponseDto;
+import mate.academy.intro.dto.OrderResponseDto;
+import mate.academy.intro.dto.UpdateOrderStatusRequestDto;
+import mate.academy.intro.mapper.OrderItemMapper;
+import mate.academy.intro.mapper.OrderMapper;
+import mate.academy.intro.model.CartItem;
+import mate.academy.intro.model.Order;
+import mate.academy.intro.model.OrderItem;
+import mate.academy.intro.model.OrderStatus;
+import mate.academy.intro.model.ShoppingCart;
+import mate.academy.intro.model.User;
+import mate.academy.intro.repository.OrderItemRepository;
+import mate.academy.intro.repository.OrderRepository;
+import mate.academy.intro.repository.ShoppingCartRepository;
+import mate.academy.intro.repository.UserRepository;
+import mate.academy.intro.service.OrderService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final UserRepository userRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository cartRepository;
+
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeOrder(Long userId, CreateOrderRequestDto requestDto) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("User not found"));
+
+        ShoppingCart cart = cartRepository.findByUserId(userId)
+                .orElseThrow(() -> new RuntimeException("Shopping cart not found"));
+
+        if (cart.getCartItems().isEmpty()) {
+            throw new RuntimeException("Shopping cart is empty");

```

Comment:
add parameter to message
```suggestion
            throw new OrderProcessinException("Shopping cart is empty");
```